src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -50,18 +54,58 @@ public static int Main(string[] args)
                     path = Directory.GetCurrentDirectory();
                 }
 
-                // Load project context and publish it
-                var fx = NuGetFramework.Parse(framework.Value());
-                var rids = new[] { runtime.Value() };
-                var context = ProjectContext.Create(path, fx, rids);
+                // get the lock file
+                var projectDir = path.ToLower().EndsWith(Project.FileName) ?
+                                    Path.GetDirectoryName(path) :
+                                    path;
+
+                var projectLockJsonPath = Path.Combine(projectDir, LockFile.FileName);
+                if (!File.Exists(projectLockJsonPath))
+                {
+                    Reporter.Output.WriteLine($"Unable to locate {LockFile.FileName} for project '{project.Value}'".Red());
+                    Reporter.Output.WriteLine($"Run 'dotnet restore' before calling 'dotnet publish'".Red());
+                    return 1;
+                }
+
+                var lockFile = LockFileReader.Read(projectLockJsonPath);
 
-                if (string.IsNullOrEmpty(context.RuntimeIdentifier))
+                var lockFileTargets = lockFile.Targets.Where(target =>
+                {
+                    if(target.TargetFramework == null || string.IsNullOrEmpty(target.RuntimeIdentifier))
+                    {
+                        return false;
+                    }
+
+                    if (!runtime.HasValue() || runtime.Value().Equals(target.RuntimeIdentifier))
+                    {
+                        if (!framework.HasValue() || NuGetFramework.Parse(framework.Value()).Equals(target.TargetFramework))
+                        {
+                            return true;
+                        }
+                    }
+
+                    return false;
+                });
+
+                if(lockFileTargets.Count() == 0)
                 {
-                    Reporter.Output.WriteLine($"Unknown runtime identifier {runtime.Value()}.".Red());
+                    string errMsg = $"'{project.Value}' cannot be published";
+                    if(framework.HasValue() || runtime.HasValue())
+                    {
+                        errMsg += $" for '{framework.Value()}' '{runtime.Value()}'";
+                    }
+
+                    Reporter.Output.WriteLine(errMsg.Red());
                     return 1;
                 }
 
-                return Publish(context, output.Value(), configuration.Value() ?? Constants.DefaultConfiguration);
+                int result = 0;
+                foreach(var target in lockFileTargets)
+                {
+                    result += Publish(target.TargetFramework, target.RuntimeIdentifier, path, output.Value(), configuration.Value() ?? Constants.DefaultConfiguration);
```

Comment:
Well thats the idea. The user can see how many publishes failed vs succeeded. This will provide information to the user on what needs to be fixed in one shot rather than running `dotnet publish` multiple times.